### DISCUSSION TITLE
ClientIdentifierNotValid

### DISCUSSION THREAD

**Discussion by rocky050371** [dotnet/MQTTnet#1832] [General]
I have the following issue when I try and introduce a Keep Alive, I tried changing the id but still no joy

"Connecting with MQTT server failed (ClientIdentifierNotValid)."

![image](https://github.com/dotnet/MQTTnet/assets/44700602/f052c4a7-a732-47e1-89e0-603327cdd501)

**Comment by SeppPenner** [comment]
Which MQTT server are you connecting to? Are there maybe some restrictions regardings the client id?
